core/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala

```diff
@@ -0,0 +1,31 @@
+package org.scalafmt.rewrite
+
+import scala.meta.Tree
+import scala.meta._
+import scala.meta.tokens.Token.{LeftParen, RightParen}
+
+object RedundantParens extends Rewrite {
+
+  def isWrappedInParens(tokens: Tokens): Boolean =
+    tokens.nonEmpty && tokens.head.is[LeftParen] && tokens.last.is[RightParen]
+
+  override def rewrite(code: Tree, ctx: RewriteCtx): Seq[Patch] = {
+    val builder = Seq.newBuilder[Patch]
+
+    code.collect {
+      case g: Enumerator.Guard =>
+        val tokens: Tokens = g.cond.tokens
+        if (isWrappedInParens(tokens)) {
+          val openParensCount = tokens.segmentLength(_.is[LeftParen], 0)
+          val closeParensCount = tokens.reverse.segmentLength(_.is[RightParen], 0)
+          val parensToRemoveCount = Math.min(openParensCount, closeParensCount)
+          val leftSegment: Tokens = tokens.slice(0, parensToRemoveCount)
+          val rightSegment: Tokens = tokens.slice(tokens.length - parensToRemoveCount, tokens.length)
```

Comment:
It is possible to takeRight.  I wasn't sure if there was value in using operations like slice that returned Tokens instead of takeRight that returned IndexedSeq[Token].  It caught me by surprise a few times when I had to change my function signature as Tokens just turned into IndexedSeq[Token].  The alias is easy to lose.